File: dagger/app/src/main/java/siarhei/luskanau/example/dagger/module/ModuleC.java
package siarhei.luskanau.example.dagger.module;

import dagger.Module;
import dagger.Provides;
import siarhei.luskanau.example.dagger.model.ModelA;
import siarhei.luskanau.example.dagger.model.ModelB;
import siarhei.luskanau.example.dagger.model.ModelC;
import siarhei.luskanau.example.dagger.scope.ScopeC;

@Module
public class ModuleC {

    @Provides
    @ScopeC
    ModelC provideModelC(ModelA modelA, ModelB modelB) {
        return new ModelC();
    }
}

File: dagger/app/src/main/java/siarhei/luskanau/example/dagger/module/ModuleC.kt
package siarhei.luskanau.example.dagger.module

import dagger.Module
import dagger.Provides
import siarhei.luskanau.example.dagger.model.ModelA
import siarhei.luskanau.example.dagger.model.ModelB
import siarhei.luskanau.example.dagger.model.ModelC
import siarhei.luskanau.example.dagger.scope.ScopeC

@Module
class ModuleC {

    @Provides
    @ScopeC
    internal fun provideModelC(modelA: ModelA, modelB: ModelB): ModelC {
        return ModelC(modelA, modelB)
    }
}
